public/app/features/integrated-alerting/components/NotificationChannel/NotificationChannel.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { FC, useMemo, useState, useEffect } from 'react';
+import { logger } from '@percona/platform-core';
+import { NotificationChannelService } from './NotificationChannel.service';
+import { Table } from '../Table/Table';
+import { NotificationChannel as Channel } from './NotificationChannel.types';
+import { Messages } from './NotificationChannel.messages';
+import { NotificationChannelProvider } from './NotificationChannel.provider';
+
+const { emptyTable, nameColumn, typeColumn, typeLabel } = Messages;
+
+export const NotificationChannel: FC = () => {
+  const [pendingRequest, setPendingRequest] = useState(false);
+  const [data, setData] = useState<Channel[]>([]);
+
+  const columns = useMemo(
+    () => [
+      {
+        Header: nameColumn,
+        accessor: 'summary',
+      },
+      {
+        Header: typeColumn,
+        accessor: ({ type }: Channel) => typeLabel[type],
+      },
+    ],
+    []
+  );
+
+  const getNotificationChannels = async () => {
+    setPendingRequest(true);
+    try {
+      setData(await NotificationChannelService.list());
+    } catch (e) {
+      logger.error(e);
+    } finally {
+      setPendingRequest(false);
+    }
+  };
+
+  useEffect(() => {
+    getNotificationChannels();
+  }, []);
+
+  return (
+    <NotificationChannelProvider.Provider value={{ getNotificationChannels }}>
```

Comment:
We had the problem in the alert rule templates that after the edit we wanted to refresh the table (and it needs to call the proper get for that). It got to some levels of passing props (like 2 or 3) which was not good and @nicolalamacchia pointed that out well. So was decided to use the context API for that (and started solving it on this PR since we will face the same issues here on notification channels). I think is makes sense since this doesn't need to be global to the whole app, this only needs to be shared among the components of the notification channel tab.

Reply:
Ok then, makes sense.